diff -ruN autopoweroff.orig/Makefile.in autopoweroff/Makefile.in
--- autopoweroff.orig/Makefile.in	2008-05-26 20:36:00.000000000 +0300
+++ autopoweroff/Makefile.in	2013-01-28 14:24:06.178344258 +0200
@@ -807,7 +807,7 @@
 # all files) are installed.
 install-data-hook:
 	echo "install-data-hook called."
-	postinstall
+	./postinstall
 #   if [ ${INSTALLTYPE} != "RPM" ]; then \
 #     if [ `id -u` != "0" ]; then \
 #       echo -e "\nThe installation can only proceed under the root user.\n"; \
