Index: network-manager-applet-0.8.3+git.20110203t003354.9bf0b98/src/applet.c
===================================================================
--- network-manager-applet-0.8.3+git.20110203t003354.9bf0b98.orig/src/applet.c	2011-02-14 17:18:54.408664077 -0500
+++ network-manager-applet-0.8.3+git.20110203t003354.9bf0b98/src/applet.c	2011-02-14 17:22:46.547351391 -0500
@@ -1649,15 +1649,15 @@
 
 	n_wireless = nma_menu_add_devices (menu, applet);
 
-	nma_menu_add_vpn_submenu (menu, applet);
-
 	if (n_wireless > 0 && nm_client_wireless_get_enabled (applet->nm_client)) {
 		/* Add the "Hidden wireless network..." entry */
-		nma_menu_add_separator_item (menu);
 		nma_menu_add_hidden_network_item (menu, applet);
 		nma_menu_add_create_network_item (menu, applet);
+		nma_menu_add_separator_item (menu);
 	}
 
+	nma_menu_add_vpn_submenu (menu, applet);
+
 	gtk_widget_show_all (menu);
 
 //	nmi_dbus_signal_user_interface_activated (applet->connection);
