From dbe3b1205185c69816c5313d7c75268ee3ba59b1 Mon Sep 17 00:00:00 2001
From: Marius B. Kotsbak <marius@kotsbak.com>
Date: Sat, 01 Sep 2012 13:48:53 +0000
Subject: applet: add support for indicating HSPA+ and LTE technologies in the applet.

---
diff --git a/src/applet-device-gsm.c b/src/applet-device-gsm.c
index 5d498aa..27e7a04 100644
--- a/src/applet-device-gsm.c
+++ b/src/applet-device-gsm.c
@@ -58,8 +58,10 @@ typedef enum {
     MM_MODEM_GSM_ACCESS_TECH_HSDPA       = 6,  /* UTRAN w/HSDPA */
     MM_MODEM_GSM_ACCESS_TECH_HSUPA       = 7,  /* UTRAN w/HSUPA */
     MM_MODEM_GSM_ACCESS_TECH_HSPA        = 8,  /* UTRAN w/HSDPA and HSUPA */
+    MM_MODEM_GSM_ACCESS_TECH_HSPA_PLUS   = 9,
+    MM_MODEM_GSM_ACCESS_TECH_LTE         = 10,
 
-    MM_MODEM_GSM_ACCESS_TECH_LAST = MM_MODEM_GSM_ACCESS_TECH_HSPA
+    MM_MODEM_GSM_ACCESS_TECH_LAST = MM_MODEM_GSM_ACCESS_TECH_LTE
 } MMModemGsmAccessTech;
 
 typedef struct {
@@ -354,6 +356,10 @@ gsm_act_to_mb_act (GsmDeviceInfo *info)
 		return MB_TECH_HSUPA;
 	case MM_MODEM_GSM_ACCESS_TECH_HSPA:
 		return MB_TECH_HSPA;
+	case MM_MODEM_GSM_ACCESS_TECH_HSPA_PLUS:
+		return MB_TECH_HSPA_PLUS;
+	case MM_MODEM_GSM_ACCESS_TECH_LTE:
+		return MB_TECH_LTE;
 	default:
 		break;
 	}
diff --git a/src/mb-menu-item.c b/src/mb-menu-item.c
index 8a32f7d..5f945eb 100644
--- a/src/mb-menu-item.c
+++ b/src/mb-menu-item.c
@@ -69,8 +69,12 @@ get_tech_name (guint32 tech)
 		return _("HSUPA");
 	case MB_TECH_HSPA:
 		return _("HSPA");
+	case MB_TECH_HSPA_PLUS:
+		return _("HSPA+");
 	case MB_TECH_WIMAX:
 		return _("WiMAX");
+	case MB_TECH_LTE:
+		return _("LTE");
 	default:
 		break;
 	}
diff --git a/src/mobile-helpers.c b/src/mobile-helpers.c
index ab5c505..756166b 100644
--- a/src/mobile-helpers.c
+++ b/src/mobile-helpers.c
@@ -117,6 +117,8 @@ mobile_helper_get_tech_icon (guint32 tech, NMApplet *applet)
 	case MB_TECH_HSDPA:
 	case MB_TECH_HSUPA:
 	case MB_TECH_HSPA:
+	case MB_TECH_HSPA_PLUS:
+	case MB_TECH_LTE:
 		return nma_icon_check_and_load ("nm-tech-hspa", &applet->mb_tech_hspa_icon, applet);
 	case MB_TECH_WIMAX:
 	default:
diff --git a/src/mobile-helpers.h b/src/mobile-helpers.h
index 28b93f9..578ec13 100644
--- a/src/mobile-helpers.h
+++ b/src/mobile-helpers.h
@@ -47,6 +47,8 @@ enum {
 	MB_TECH_HSDPA,
 	MB_TECH_HSUPA,
 	MB_TECH_HSPA,
+	MB_TECH_HSPA_PLUS,
+	MB_TECH_LTE,
 	MB_TECH_WIMAX,
 };
 
--
cgit v0.9.0.2
