=== modified file 'src/applet-device-wifi.c'
---
 src/applet-device-wifi.c |    9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

Index: b/src/applet-device-wifi.c
===================================================================
--- a/src/applet-device-wifi.c
+++ b/src/applet-device-wifi.c
@@ -1245,7 +1245,6 @@ wifi_device_state_changed (NMDevice *dev
                            NMApplet *applet)
 {
 	NMAccessPoint *new = NULL;
-	char *msg;
 	char *esc_ssid = NULL;
 
 	new = update_active_ap (device, new_state, applet);
@@ -1257,11 +1256,11 @@ wifi_device_state_changed (NMDevice *dev
 		return;
 
 	esc_ssid = get_ssid_utf8 (new);
-	msg = g_strdup_printf (_("You are now connected to the Wi-Fi network '%s'."), esc_ssid);
-	applet_do_notify_with_pref (applet, _("Connection Established"),
-	                            msg, "nm-device-wireless",
+	applet_do_notify_with_pref (applet,
+	                            _("Connected"),
+	                            esc_ssid ? esc_ssid : _("(none)"),
+	                            "nm-device-wireless",
 	                            PREF_DISABLE_CONNECTED_NOTIFICATIONS);
-	g_free (msg);
 	g_free (esc_ssid);
 }
 
