---
 src/applet.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: network-manager-applet-0.8.999+git.20110523t134536.bb7ef97/src/applet.c
===================================================================
--- network-manager-applet-0.8.999+git.20110523t134536.bb7ef97.orig/src/applet.c	2011-05-26 11:44:24.000000000 -0400
+++ network-manager-applet-0.8.999+git.20110523t134536.bb7ef97/src/applet.c	2011-05-26 11:58:42.954951054 -0400
@@ -2230,7 +2230,7 @@
 				info->icon = "notification-gsm-disconnected";
 				break;
 			default:
-				info->icon = "nm-no-connection";
+				info->icon = "notification-network-disconnected";
 				break;
 		}
 		g_debug("going for offline with icon: %s", info->icon);
