--- gnome-session-3.9.90.orig/gnome-session/gsm-manager.c
+++ gnome-session-3.9.90/gnome-session/gsm-manager.c
@@ -86,7 +86,8 @@
  * a fail whale if required components don't show up quickly enough,
  * let's make this fairly long.
  */
-#define GSM_MANAGER_PHASE_TIMEOUT 90 /* seconds */
+#define GSM_MANAGER_PHASE_TIMEOUT      90 /* seconds */
+#define GSM_MANAGER_PHASE_EXIT_TIMEOUT 4  /* seconds */
 
 #define GDM_FLEXISERVER_COMMAND "gdmflexiserver"
 #define GDM_FLEXISERVER_ARGS    "--startnew Standard"
@@ -893,7 +894,7 @@ do_phase_end_session (GsmManager *manage
         }
 
         if (gsm_store_size (manager->priv->clients) > 0) {
-                manager->priv->phase_timeout_id = g_timeout_add_seconds (GSM_MANAGER_PHASE_TIMEOUT,
+                manager->priv->phase_timeout_id = g_timeout_add_seconds (GSM_MANAGER_PHASE_EXIT_TIMEOUT,
                                                                          (GSourceFunc)on_phase_timeout,
                                                                          manager);
 
@@ -961,7 +962,7 @@ static void
 do_phase_exit (GsmManager *manager)
 {
         if (gsm_store_size (manager->priv->clients) > 0) {
-                manager->priv->phase_timeout_id = g_timeout_add_seconds (GSM_MANAGER_PHASE_TIMEOUT,
+                manager->priv->phase_timeout_id = g_timeout_add_seconds (GSM_MANAGER_PHASE_EXIT_TIMEOUT,
                                                                          (GSourceFunc)on_phase_timeout,
                                                                          manager);
 
