Index: unity-control-center/panels/info/info.ui
===================================================================
--- unity-control-center.orig/panels/info/info.ui	2014-04-10 17:26:50.000000000 +0300
+++ unity-control-center/panels/info/info.ui	2014-09-08 12:19:44.046218304 +0300
@@ -1,6 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!-- Generated with glade 3.16.1 -->
 <interface>
-  <!-- interface-requires gtk+ 3.0 -->
+  <requires lib="gtk+" version="3.0"/>
   <object class="GtkDialog" id="extra_options_dialog">
     <property name="can_focus">False</property>
     <property name="border_width">10</property>
@@ -25,12 +26,12 @@
             <child>
               <object class="GtkButton" id="button1">
                 <property name="label">gtk-close</property>
+                <property name="use_action_appearance">False</property>
                 <property name="visible">True</property>
                 <property name="can_focus">True</property>
                 <property name="can_default">True</property>
                 <property name="has_default">True</property>
                 <property name="receives_default">True</property>
-                <property name="use_action_appearance">False</property>
                 <property name="use_stock">True</property>
               </object>
               <packing>
@@ -57,8 +58,8 @@
               <object class="GtkBox" id="vbox50">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
-                <property name="spacing">10</property>
                 <property name="orientation">vertical</property>
+                <property name="spacing">10</property>
                 <child>
                   <object class="GtkLabel" id="label61">
                     <property name="visible">True</property>
@@ -84,14 +85,14 @@
                       <object class="GtkBox" id="vbox51">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
-                        <property name="spacing">6</property>
                         <property name="orientation">vertical</property>
+                        <property name="spacing">6</property>
                         <child>
                           <object class="GtkGrid" id="table5">
                             <property name="visible">True</property>
                             <property name="can_focus">False</property>
-                            <property name="column_spacing">6</property>
                             <property name="row_spacing">10</property>
+                            <property name="column_spacing">6</property>
                             <child>
                               <object class="GtkComboBox" id="media_other_type_combobox">
                                 <property name="visible">True</property>
@@ -100,6 +101,8 @@
                               <packing>
                                 <property name="left_attach">1</property>
                                 <property name="top_attach">0</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -113,6 +116,8 @@
                               <packing>
                                 <property name="left_attach">0</property>
                                 <property name="top_attach">1</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -123,6 +128,8 @@
                               <packing>
                                 <property name="left_attach">1</property>
                                 <property name="top_attach">1</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -137,6 +144,8 @@
                               <packing>
                                 <property name="left_attach">0</property>
                                 <property name="top_attach">0</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                           </object>
@@ -188,7 +197,6 @@
             <property name="visible">True</property>
             <property name="can_focus">False</property>
             <property name="spacing">10</property>
-            <property name="orientation">horizontal</property>
             <child>
               <object class="GtkScrolledWindow" id="scrolledwindow1">
                 <property name="visible">True</property>
@@ -216,8 +224,8 @@
               <object class="GtkBox" id="detail_vbox">
                 <property name="visible">True</property>
                 <property name="can_focus">False</property>
-                <property name="spacing">20</property>
                 <property name="orientation">vertical</property>
+                <property name="spacing">20</property>
                 <child>
                   <object class="GtkNotebook" id="notebook">
                     <property name="visible">True</property>
@@ -233,8 +241,8 @@
                           <object class="GtkBox" id="vbox2">
                             <property name="visible">True</property>
                             <property name="can_focus">False</property>
-                            <property name="spacing">18</property>
                             <property name="orientation">vertical</property>
+                            <property name="spacing">18</property>
                             <child>
                               <object class="GtkImage" id="system_image">
                                 <property name="visible">True</property>
@@ -349,10 +357,8 @@
                                 </child>
                                 <child>
                                   <object class="GtkEntry" id="name_entry">
-                                    <property name="visible">False</property>
-                                    <property name="xalign">0</property>
-                                    <property name="text"></property>
                                     <property name="sensitive">False</property>
+                                    <property name="can_focus">False</property>
                                   </object>
                                   <packing>
                                     <property name="left_attach">1</property>
@@ -538,37 +544,6 @@
                                 <property name="position">2</property>
                               </packing>
                             </child>
-                            <child>
-                              <object class="GtkButtonBox" id="hbuttonbox1">
-                                <property name="visible">True</property>
-                                <property name="can_focus">False</property>
-                                <property name="layout_style">end</property>
-                                <property name="orientation">horizontal</property>
-                                <child>
-                                  <placeholder/>
-                                </child>
-                                <child>
-                                  <object class="GtkButton" id="updates_button">
-                                    <property name="label">Updates Available</property>
-                                    <property name="can_focus">True</property>
-                                    <property name="receives_default">True</property>
-                                    <property name="use_action_appearance">False</property>
-                                    <property name="visible">True</property>
-                                  </object>
-                                  <packing>
-                                    <property name="expand">False</property>
-                                    <property name="fill">True</property>
-                                    <property name="position">1</property>
-                                  </packing>
-                                </child>
-                              </object>
-                              <packing>
-                                <property name="expand">False</property>
-                                <property name="fill">False</property>
-                                <property name="pack_type">end</property>
-                                <property name="position">3</property>
-                              </packing>
-                            </child>
                           </object>
                         </child>
                       </object>
@@ -590,14 +565,14 @@
                         <property name="halign">center</property>
                         <property name="valign">start</property>
                         <property name="border_width">20</property>
-                        <property name="spacing">10</property>
                         <property name="orientation">vertical</property>
+                        <property name="spacing">10</property>
                         <child>
                           <object class="GtkGrid" id="default_apps_grid">
                             <property name="visible">True</property>
                             <property name="can_focus">False</property>
-                            <property name="column_spacing">12</property>
                             <property name="row_spacing">12</property>
+                            <property name="column_spacing">12</property>
                             <child>
                               <object class="GtkLabel" id="web-label">
                                 <property name="visible">True</property>
@@ -609,6 +584,12 @@
                                   <class name="dim-label-ucc"/>
                                 </style>
                               </object>
+                              <packing>
+                                <property name="left_attach">0</property>
+                                <property name="top_attach">0</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
+                              </packing>
                             </child>
                             <child>
                               <object class="GtkLabel" id="mail-label">
@@ -622,8 +603,10 @@
                                 </style>
                               </object>
                               <packing>
-                                <property name="top_attach">1</property>
                                 <property name="left_attach">0</property>
+                                <property name="top_attach">1</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -638,8 +621,10 @@
                                 </style>
                               </object>
                               <packing>
-                                <property name="top_attach">2</property>
                                 <property name="left_attach">0</property>
+                                <property name="top_attach">2</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -654,8 +639,10 @@
                                 </style>
                               </object>
                               <packing>
-                                <property name="top_attach">3</property>
                                 <property name="left_attach">0</property>
+                                <property name="top_attach">3</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -670,8 +657,10 @@
                                 </style>
                               </object>
                               <packing>
-                                <property name="top_attach">4</property>
                                 <property name="left_attach">0</property>
+                                <property name="top_attach">4</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -683,6 +672,8 @@
                               <packing>
                                 <property name="left_attach">2</property>
                                 <property name="top_attach">0</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -694,6 +685,8 @@
                               <packing>
                                 <property name="left_attach">2</property>
                                 <property name="top_attach">1</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -705,6 +698,8 @@
                               <packing>
                                 <property name="left_attach">2</property>
                                 <property name="top_attach">2</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -716,6 +711,8 @@
                               <packing>
                                 <property name="left_attach">2</property>
                                 <property name="top_attach">3</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -727,6 +724,8 @@
                               <packing>
                                 <property name="left_attach">2</property>
                                 <property name="top_attach">4</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -741,8 +740,10 @@
                                 </style>
                               </object>
                               <packing>
-                                <property name="top_attach">5</property>
                                 <property name="left_attach">0</property>
+                                <property name="top_attach">5</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -754,6 +755,8 @@
                               <packing>
                                 <property name="left_attach">2</property>
                                 <property name="top_attach">5</property>
+                                <property name="width">1</property>
+                                <property name="height">1</property>
                               </packing>
                             </child>
                             <child>
@@ -801,7 +804,6 @@
                       <object class="GtkBox" id="media_preferences_vbox">
                         <property name="visible">True</property>
                         <property name="can_focus">False</property>
-                        <property name="orientation">horizontal</property>
                         <child>
                           <object class="GtkAlignment" id="alignment1">
                             <property name="visible">True</property>
@@ -811,22 +813,22 @@
                                 <property name="visible">True</property>
                                 <property name="can_focus">False</property>
                                 <property name="border_width">10</property>
-                                <property name="spacing">10</property>
                                 <property name="orientation">vertical</property>
+                                <property name="spacing">10</property>
                                 <child>
                                   <object class="GtkBox" id="media_handling_vbox">
                                     <property name="visible">True</property>
                                     <property name="can_focus">False</property>
-                                    <property name="spacing">10</property>
                                     <property name="valign">start</property>
                                     <property name="vexpand">False</property>
                                     <property name="orientation">vertical</property>
+                                    <property name="spacing">10</property>
                                     <child>
                                       <object class="GtkBox" id="vbox44">
                                         <property name="visible">True</property>
                                         <property name="can_focus">False</property>
-                                        <property name="spacing">10</property>
                                         <property name="orientation">vertical</property>
+                                        <property name="spacing">10</property>
                                         <child>
                                           <object class="GtkLabel" id="label42">
                                             <property name="visible">True</property>
@@ -852,8 +854,8 @@
                                               <object class="GtkBox" id="vbox52">
                                                 <property name="visible">True</property>
                                                 <property name="can_focus">False</property>
-                                                <property name="spacing">6</property>
                                                 <property name="orientation">vertical</property>
+                                                <property name="spacing">6</property>
                                                 <child>
                                                   <object class="GtkTable" id="table4">
                                                     <property name="visible">True</property>
@@ -876,7 +878,7 @@
                                                       </object>
                                                       <packing>
                                                         <property name="x_options">GTK_FILL</property>
-                                                        <property name="y_options"></property>
+                                                        <property name="y_options"/>
                                                       </packing>
                                                     </child>
                                                     <child>
@@ -895,7 +897,7 @@
                                                         <property name="top_attach">1</property>
                                                         <property name="bottom_attach">2</property>
                                                         <property name="x_options">GTK_FILL</property>
-                                                        <property name="y_options"></property>
+                                                        <property name="y_options"/>
                                                       </packing>
                                                     </child>
                                                     <child>
@@ -941,7 +943,7 @@
                                                         <property name="top_attach">2</property>
                                                         <property name="bottom_attach">3</property>
                                                         <property name="x_options">GTK_FILL</property>
-                                                        <property name="y_options"></property>
+                                                        <property name="y_options"/>
                                                       </packing>
                                                     </child>
                                                     <child>
@@ -975,7 +977,7 @@
                                                         <property name="top_attach">3</property>
                                                         <property name="bottom_attach">4</property>
                                                         <property name="x_options">GTK_FILL</property>
-                                                        <property name="y_options"></property>
+                                                        <property name="y_options"/>
                                                       </packing>
                                                     </child>
                                                     <child>
@@ -1009,7 +1011,7 @@
                                                         <property name="top_attach">4</property>
                                                         <property name="bottom_attach">5</property>
                                                         <property name="x_options">GTK_FILL</property>
-                                                        <property name="y_options"></property>
+                                                        <property name="y_options"/>
                                                       </packing>
                                                     </child>
                                                     <child>
@@ -1038,14 +1040,13 @@
                                                   <object class="GtkBox" id="hbox2">
                                                     <property name="visible">True</property>
                                                     <property name="can_focus">False</property>
-                                                    <property name="orientation">horizontal</property>
                                                     <child>
                                                       <object class="GtkButton" id="extra_options_button">
                                                         <property name="label" translatable="yes">_Other Media…</property>
+                                                        <property name="use_action_appearance">False</property>
                                                         <property name="visible">True</property>
                                                         <property name="can_focus">True</property>
                                                         <property name="receives_default">True</property>
-                                                        <property name="use_action_appearance">False</property>
                                                         <property name="use_underline">True</property>
                                                       </object>
                                                       <packing>
@@ -1088,15 +1089,15 @@
                                 <child>
                                   <object class="GtkCheckButton" id="media_autorun_never_checkbutton">
                                     <property name="label" translatable="yes">_Never prompt or start programs on media insertion</property>
+                                    <property name="use_action_appearance">False</property>
                                     <property name="visible">True</property>
                                     <property name="can_focus">True</property>
                                     <property name="receives_default">False</property>
-                                    <property name="use_action_appearance">False</property>
+                                    <property name="valign">end</property>
+                                    <property name="vexpand">False</property>
                                     <property name="use_underline">True</property>
                                     <property name="xalign">0</property>
                                     <property name="draw_indicator">True</property>
-                                    <property name="valign">end</property>
-                                    <property name="vexpand">False</property>
                                   </object>
                                   <packing>
                                     <property name="expand">False</property>
@@ -1130,10 +1131,10 @@
                       </packing>
                     </child>
                     <child>
-					  <placeholder/>
+                      <placeholder/>
                     </child>
                     <child type="tab">
-					  <placeholder/>
+                      <placeholder/>
                     </child>
                   </object>
                   <packing>
Index: unity-control-center/panels/info/cc-info-panel.c
===================================================================
--- unity-control-center.orig/panels/info/cc-info-panel.c	2014-04-10 17:26:50.000000000 +0300
+++ unity-control-center/panels/info/cc-info-panel.c	2014-09-08 12:20:35.682466961 +0300
@@ -1648,30 +1648,7 @@
 static void
 refresh_update_button (CcInfoPanel  *self)
 {
-  GtkWidget *widget;
-
-  widget = WID ("updates_button");
-  if (widget == NULL)
-    return;
-
-  switch (self->priv->updates_state)
-    {
-      case PK_NOT_AVAILABLE:
-        gtk_widget_set_visible (widget, FALSE);
-        break;
-      case UPDATES_AVAILABLE:
-        gtk_widget_set_sensitive (widget, TRUE);
-        gtk_button_set_label (GTK_BUTTON (widget), _("Install Updates"));
-        break;
-      case UPDATES_NOT_AVAILABLE:
-        gtk_widget_set_sensitive (widget, FALSE);
-        gtk_button_set_label (GTK_BUTTON (widget), _("System Up-To-Date"));
-        break;
-      case CHECKING_UPDATES:
-        gtk_widget_set_sensitive (widget, FALSE);
-        gtk_button_set_label (GTK_BUTTON (widget), _("Checking for Updates"));
-        break;
-    }
+  return;
 }
 
 static void
@@ -1926,9 +1903,6 @@
 
   self->priv->graphics_data = get_graphics_data ();
 
-  widget = WID ("updates_button");
-  g_signal_connect (widget, "clicked", G_CALLBACK (on_updates_button_clicked), self);
-
   info_panel_setup_updates (self);
   info_panel_setup_selector (self);
   info_panel_setup_overview (self);
