From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: Force a menu update 30 seconds after the initialization is done.

---
 src/applet.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

Index: network-manager-applet/src/applet.c
===================================================================
--- network-manager-applet.orig/src/applet.c	2014-07-15 15:25:41.945414059 +0300
+++ network-manager-applet/src/applet.c	2014-07-15 15:25:41.941414044 +0300
@@ -3514,6 +3514,8 @@
 	if (!applet->context_menu)
 		return FALSE;
 
+	g_timeout_add_seconds (30, (GSourceFunc) applet_update_indicator_menu, applet);
+
 	return TRUE;
 }
 
