From: Guido Guenther <agx@sigxcpu.org>
Date: Fri, 7 Mar 2008 09:08:25 +0100
Subject: remove-RHism.diff

---
 tools/virsh.pod |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: libvirt-1.0.1/tools/virsh.pod
===================================================================
--- libvirt-1.0.1.orig/tools/virsh.pod	2013-01-22 10:43:01.404737401 -0600
+++ libvirt-1.0.1/tools/virsh.pod	2013-01-22 10:43:01.396737401 -0600
@@ -107,7 +107,7 @@
 
 Most B<virsh> operations rely upon the libvirt library being able to
 connect to an already running libvirtd service.  This can usually be
-done using the command B<service libvirtd start>.
+done using the command B<invoke-rc.d libvirt-bin start>.
 
 Most B<virsh> commands require root privileges to run due to the
 communications channels used to talk to the hypervisor.  Running as
