Description: Fix expres{,s}ion typo
Author: Didier Raboud <odyx@debian.org>
Origin: vendor
Bug: https://cups.org/str.php?L4350
Last-Update: 2014-01-04
--- a/cgi-bin/template.c
+++ b/cgi-bin/template.c
@@ -553,7 +553,7 @@
 	      if (regcomp(&re, compare, REG_EXTENDED | REG_ICASE))
 	      {
 	        fprintf(stderr,
-		        "ERROR: Unable to compile regular expresion \"%s\"!\n",
+		        "ERROR: Unable to compile regular expression \"%s\"!\n",
 			compare);
 		result = 0;
 	      }
