Description: Drop dangling references from cups-lpd.man
Author: Bastien ROUCARIÈS <bastien.roucaries@u-cergy.fr>
Bug-Debian: http://bugs.debian.org/570157
Last-Update: 2013-03-14
--- a/man/cups-lpd.man.in
+++ b/man/cups-lpd.man.in
@@ -113,7 +113,7 @@
 stray from this definition, remote status reporting to LPD
 clients may be unreliable.
 .SH SEE ALSO
-\fIcups(1)\fR, \fIcupsd(8)\fR, \fIinetconv(1m)\fR,
+\fIcupsd(8)\fR,
 \fIinetd(8)\fR, \fIxinetd(8)\fR,
 .br
 http://localhost:631/help
