Index: xserver-xorg-input-synaptics.git/src/synaptics.c
===================================================================
--- xserver-xorg-input-synaptics.git.orig/src/synaptics.c	2013-12-10 13:41:57.754590117 +0100
+++ xserver-xorg-input-synaptics.git/src/synaptics.c	2013-12-10 13:41:57.754590117 +0100
@@ -2466,7 +2466,9 @@
     uint32_t close_point = 0; /* 1 bit for each point close to another one */
     int i, j;
 
+#ifdef BUG_RETURN_VAL
     BUG_RETURN_VAL(hw->num_mt_mask > sizeof(close_point) * 8, 0);
+#endif
 
     for (i = 0; i < hw->num_mt_mask - 1; i++) {
         ValuatorMask *f1;
