--- goobox_original/goobox/src/goo-window.c	2014-09-10 10:05:02.217027998 +0300
+++ goobox/src/goo-window.c	2014-09-24 10:44:47.455546784 +0300
@@ -309,14 +309,14 @@
 				   max_window_height);
 	}
 	else {
-		hints.max_height = -1;
+		hints.max_height = G_MAXINT;
 		hints.max_width = G_MAXINT;
 		gtk_window_set_geometry_hints (GTK_WINDOW (window),
 					       GTK_WIDGET (window),
 					       &hints,
 					       GDK_HINT_MAX_SIZE);
 
-		window->priv->resizable_playlist = FALSE;
+		window->priv->resizable_playlist = TRUE;
 	}
 
 	gtk_scrolled_window_set_policy (GTK_SCROLLED_WINDOW (window->priv->list_scrolled_window),
