diff -urN redis-2.8.13-orig/src/Makefile redis-2.8.13/src/Makefile
--- redis-2.8.13-orig/src/Makefile	2014-07-14 08:48:14.000000000 -0700
+++ redis-2.8.13/src/Makefile	2014-07-20 15:19:40.575056196 -0700
@@ -84,8 +84,8 @@
 
 ifeq ($(MALLOC),jemalloc)
 	DEPENDENCY_TARGETS+= jemalloc
-	FINAL_CFLAGS+= -DUSE_JEMALLOC -I../deps/jemalloc/include
-	FINAL_LIBS+= ../deps/jemalloc/lib/libjemalloc.a -ldl
+	FINAL_CFLAGS+= -DUSE_JEMALLOC -I/usr/include/jemalloc/include
+	FINAL_LIBS+= -ljemalloc
 endif
 
 REDIS_CC=$(QUIET_CC)$(CC) $(FINAL_CFLAGS)
diff -urN redis-2.8.13-orig/src/zmalloc.c redis-2.8.13/src/zmalloc.c
--- redis-2.8.13-orig/src/zmalloc.c	2014-07-14 08:48:14.000000000 -0700
+++ redis-2.8.13/src/zmalloc.c	2014-07-20 15:20:48.815801388 -0700
@@ -61,10 +61,10 @@
 #define realloc(ptr,size) tc_realloc(ptr,size)
 #define free(ptr) tc_free(ptr)
 #elif defined(USE_JEMALLOC)
-#define malloc(size) je_malloc(size)
-#define calloc(count,size) je_calloc(count,size)
-#define realloc(ptr,size) je_realloc(ptr,size)
-#define free(ptr) je_free(ptr)
+#define malloc(size) malloc(size)
+#define calloc(count,size) calloc(count,size)
+#define realloc(ptr,size) realloc(ptr,size)
+#define free(ptr) free(ptr)
 #endif
 
 #if defined(__ATOMIC_RELAXED)
diff -urN redis-2.8.13-orig/src/zmalloc.h redis-2.8.13/src/zmalloc.h
--- redis-2.8.13-orig/src/zmalloc.h	2014-07-14 08:48:14.000000000 -0700
+++ redis-2.8.13/src/zmalloc.h	2014-07-20 15:21:20.508147495 -0700
@@ -50,7 +50,7 @@
 #include <jemalloc/jemalloc.h>
 #if (JEMALLOC_VERSION_MAJOR == 2 && JEMALLOC_VERSION_MINOR >= 1) || (JEMALLOC_VERSION_MAJOR > 2)
 #define HAVE_MALLOC_SIZE 1
-#define zmalloc_size(p) je_malloc_usable_size(p)
+#define zmalloc_size(p) malloc_usable_size(p)
 #else
 #error "Newer version of jemalloc required"
 #endif
