Index: xf86-input-wacom/conf/50-wacom.conf
===================================================================
--- xf86-input-wacom.orig/conf/50-wacom.conf	2011-11-17 11:05:39.017966123 +0200
+++ xf86-input-wacom/conf/50-wacom.conf	2011-11-17 11:06:15.843084804 +0200
@@ -1,6 +1,6 @@
 Section "InputClass"
 	Identifier "Wacom class"
-	MatchProduct "Wacom|WACOM|Hanwang|PTK-540WL"
+	MatchProduct "Wacom|WACOM|Hanwang|PTK-540WL|ISD-V4"
 	MatchDevicePath "/dev/input/event*"
 	Driver "wacom"
 EndSection
