diff -ruN autopoweroff.orig/etc/autopoweroff.conf autopoweroff/etc/autopoweroff.conf
--- autopoweroff.orig/etc/autopoweroff.conf	2008-06-02 01:50:16.000000000 +0300
+++ autopoweroff/etc/autopoweroff.conf	2013-01-28 14:59:36.106432054 +0200
@@ -13,9 +13,9 @@
 #   and EndHour=22, the computer will not shut down between 05:00 and
 #   22:00, local time.
 
-[NO_SHUTDOWN_TIME_RANGE]
-StartHour=5
-EndHour=22
+# [NO_SHUTDOWN_TIME_RANGE]
+# StartHour=5
+# EndHour=22
 
 
 # StartupDelay parameter (expressed in minutes):
@@ -43,9 +43,9 @@
 #   write to me).  Unfortunatly, USB keyboards and mices are not
 #   supported yet (I do not know how to detect activity on USB devices).
 
-[TIMEOUTS]
-StartupDelay=0
-IdleTime=10
+# [TIMEOUTS]
+# StartupDelay=0
+# IdleTime=10
 
 
 # Hosts parameter (list of hostnames or IPs, separated by commas):
@@ -54,5 +54,5 @@
 #   computer should not shutdown if any of the hosts declared here is
 #   still up (responding to ping).
 
-[DEPENDANTS]
-Hosts=mafalda, 192.168.0.5
+# [DEPENDANTS]
+# Hosts=mafalda, 192.168.0.5
