diff -ruN autopoweroff.orig/Makefile.am autopoweroff/Makefile.am
--- autopoweroff.orig/Makefile.am	2008-05-26 20:35:33.000000000 +0300
+++ autopoweroff/Makefile.am	2013-01-28 15:31:50.626518015 +0200
@@ -24,9 +24,6 @@
 	sbin/autopoweroff-upgrade.in \
 	python/modules/apodoc.py \
 	python/modules/apoconfig.py \
-	etc/init.d/autopoweroff.lsb \
-	etc/init.d/autopoweroff.rc-status \
-	etc/init.d/autopoweroff.redhat \
 	glade/autopoweroff.glade \
 	glade/autopoweroff.gladep \
 	glade/autopoweroff.trans \
@@ -70,10 +67,7 @@
 security_DATA = etc/security/console.apps/autopoweroff
 
 autoinitdir = $(datadir)/init.d
-autoinit_SCRIPTS = \
-  etc/init.d/autopoweroff.lsb \
-  etc/init.d/autopoweroff.rc-status \
-  etc/init.d/autopoweroff.redhat
+autoinit_SCRIPTS =
 
 # No need to setup initdir = ... here.  This variable is setup automatically
 # by configure.in  If you set it up here, you will cause problems.
