Index: gvfs-1.14.0/client/gvfsfusedaemon.c
===================================================================
--- gvfs-1.14.0.orig/client/gvfsfusedaemon.c	2012-10-01 16:27:52.911158075 -0400
+++ gvfs-1.14.0/client/gvfsfusedaemon.c	2012-10-01 16:32:20.011160901 -0400
@@ -379,7 +379,7 @@
   mount_record = g_new (MountRecord, 1);
   
   mount_record->root = g_mount_get_root (mount);
-  mount_record->name = g_object_get_data (G_OBJECT (mount), "g-stable-name");
+  mount_record->name = g_strdup (g_object_get_data (G_OBJECT (mount), "g-stable-name"));
   mount_record->creation_time = time (NULL);
   
   return mount_record;
