Index: ktouch/src/ktouch.kcfg
===================================================================
--- ktouch.orig/src/ktouch.kcfg	2014-07-01 10:28:14.000000000 +0300
+++ ktouch/src/ktouch.kcfg	2015-04-23 13:41:17.503785110 +0300
@@ -29,14 +29,14 @@
     </entry>
     <entry name="RequiredStrokesPerMinute" type="Int">
       <label>The minimum required amount of strokes per minute to unlock the next lesson.</label>
-      <default>180</default>
+      <default>30</default>
       <min>0</min>
       <max>360</max>
     </entry>
     <entry name="RequiredAccuracy" type="Double">
       <label>The minimum required typing accuracy in percent to unlock the next lesson.</label>
-      <default>98</default>
-      <min>90</min>
+      <default>50</default>
+      <min>0</min>
       <max>100</max>
     </entry>
   </group>
