Index: nfs-utils/utils/gssd/gssd_proc.c
===================================================================
--- nfs-utils.orig/utils/gssd/gssd_proc.c       2014-05-21 16:02:47.995399101 +0300
+++ nfs-utils/utils/gssd/gssd_proc.c    2014-05-21 16:04:11.823820964 +0300
@@ -696,7 +696,7 @@
        buf_size = sizeof(uid) + sizeof(timeout) + sizeof(pd->pd_seq_win) +
                sizeof(pd->pd_ctx_hndl.length) + pd->pd_ctx_hndl.length +
                sizeof(context_token->length) + context_token->length +
-               acceptor->length;
+               sizeof(acceptor->length) + acceptor->length;
        p = buf = malloc(buf_size);
        if (!buf)
                goto out_err;
