diff -ruN indicator-applet-12.10.1.orig/src/applet-main.c indicator-applet-12.10.1/src/applet-main.c
--- indicator-applet-12.10.1.orig/src/applet-main.c	2012-09-19 19:17:44.000000000 +0300
+++ indicator-applet-12.10.1/src/applet-main.c	2013-03-04 16:43:15.915148786 +0200
@@ -607,6 +607,30 @@
 
 
 static gboolean
+load_module_is_ok (const gchar * name)
+{
+  char *ia_substr, *indicator_applets, *test_module_name;
+
+  indicator_applets = getenv("HIDE_INDICATOR_APPLETS");
+  if (indicator_applets) {
+    ia_substr = indicator_applets = strdup(indicator_applets);
+    if (ia_substr) {
+      while ((test_module_name = strsep(&ia_substr, ":")) != NULL) {
+        if (strcmp(test_module_name, name) == 0) {
+          /* this module should not be shown */
+          free(indicator_applets);
+          return FALSE;
+        }
+      }
+      free(indicator_applets);
+    }
+  }
+
+  /* show this module unless environment variables show otherwise */
+  return TRUE;
+}
+
+static gboolean
 load_module (const gchar * name, GtkWidget * menubar)
 {
   GObject * o;
@@ -618,6 +642,10 @@
     return FALSE;
   }
 
+  if (!load_module_is_ok(name)) {
+    return FALSE;
+  }
+
   g_debug("Loading Module: %s", name);
 
   /* Build the object for the module */
